pandas/core/arrays/integer.py

```diff
@@ -147,6 +147,11 @@ def coerce_to_array(values, dtype, mask=None, copy=False):
             dtype = values.dtype
 
     if dtype is not None:
+        if (isinstance(dtype, string_types) and
+                (dtype.startswith("Int") or dtype.startswith("UInt"))):
+            # Avoid DeprecationWarning from NumPy about np.dtype("Int64")
+            # https://github.com/numpy/numpy/pull/7476
+            dtype = dtype.lower()
```

Comment:
@jreback could you take a close look here. I think you ran into this when writing a test.

Reply:
This is a deprecation warning from NumPy right now, which is filtered interactive, but pytest removes that filter by default. Some of the `integer_array` tests were hitting this.

Reply:
oh I didn't know numpy did this:

```
In [1]: np.dtype('Int64')
/Users/jreback/miniconda3/envs/pandas/bin/ipython:1: DeprecationWarning: Numeric-style type codes are deprecated and will result in an error in the future.
  #!/Users/jreback/miniconda3/envs/pandas/bin/python
Out[1]: dtype('int64')
```

I guess so, thanks

Reply:
Yeah. I'm waiting to hear back on https://github.com/numpy/numpy/pull/7476#issuecomment-422358051

If `np.dtype("int") == "Int64"` errors in the future, we may have to rework some things.